File: Application/Classes/Base/BaseNavigationController.m
//
//  BaseNavigationController.m
//  NikmeApps
//
//  Created by Linh NGUYEN on 9/5/14.
//  Copyright (c) 2014 Nikmesoft Ltd. All rights reserved.
//

#import "BaseNavigationController.h"

@interface BaseNavigationController ()

@end

@implementation BaseNavigationController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self.navigationBar setTranslucent:NO];
    [self.navigationBar setTintColor:[UIColor whiteColor]];
    [[UIBarButtonItem appearance] setBackButtonTitlePositionAdjustment:UIOffsetMake(0,-70)                                            forBarMetrics:UIBarMetricsDefault];
    [self.navigationBar setTitleTextAttributes:@{NSForegroundColorAttributeName : [UIColor whiteColor],
                                                 NSFontAttributeName : [UIFont fontWithName:@"Raleway-Regular" size:22]}];
    
    [self.navigationBar setBarTintColor:KNAV_BG_COLOR];
}

@end

File: CoAssetsApps/COCore/Classes/Base/BaseNavigationController.swift
//
//  BaseNavigationController.swift
//  CoAssets-Agent
//
//  Created by Linh NGUYEN on 11/24/15.
//  Copyright © 2015 Nikmesoft Ltd. All rights reserved.
//

import UIKit

class BaseNavigationController: UINavigationController, UIGestureRecognizerDelegate, UINavigationControllerDelegate {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        if self.respondsToSelector("interactivePopGestureRecognizer") {
            self.interactivePopGestureRecognizer?.delegate = self
            self.interactivePopGestureRecognizer?.enabled = false
            if let slideMenu = kAppDelegate?.window?.rootViewController as? ExSlideMenuController {
                if slideMenu.mainViewController == self {
                    slideMenu.lockSwipe = false
                }
            }
            self.delegate = self
        }
        self.navigationBar.translucent = true
        self.navigationBar.tintColor = UIColor.blackColor()
        self.navigationBar.titleTextAttributes = [NSFontAttributeName: UIFont(name: AppDefine.AppFontName.COAvenirRoman, size: 17.5*AppDefine.ScreenSize.ScreenScale)!]
    }
    
    override func pushViewController(viewController: UIViewController, animated: Bool) {
        if self.respondsToSelector("interactivePopGestureRecognizer") {
            self.interactivePopGestureRecognizer?.enabled = false
        }
        if let slideMenu = kAppDelegate?.window?.rootViewController as? ExSlideMenuController {
            if slideMenu.mainViewController == self {
                slideMenu.lockSwipe = false
            }
        }
        super.pushViewController(viewController, animated:animated)
    }
    
    func navigationController(navigationController: UINavigationController, didShowViewController viewController: UIViewController, animated: Bool) {
        if viewController == self.viewControllers.first || self.viewControllers.first!.isKindOfClass(LoginViewController.classForCoder()) {
            if self.respondsToSelector("interactivePopGestureRecognizer") {
                self.interactivePopGestureRecognizer?.enabled = false
            }
            if let slideMenu = kAppDelegate?.window?.rootViewController as? ExSlideMenuController {
                if slideMenu.mainViewController == self {
                    slideMenu.lockSwipe = false
                }
            }
        } else {
            if self.respondsToSelector("interactivePopGestureRecognizer") {
                self.interactivePopGestureRecognizer?.enabled = true
            }
            if let slideMenu = kAppDelegate?.window?.rootViewController as? ExSlideMenuController {
                if slideMenu.mainViewController == self {
                    slideMenu.lockSwipe = true
                }
            }
        }
    }
    
    func navigationController(navigationController: UINavigationController, willShowViewController viewController: UIViewController, animated: Bool) {
        if viewController == self.viewControllers.first {
            // nothing
        } else {
            
        }
    }
    
}
